src/app/pages/storage2/modules/devices/components/devices/devices.component.html

```diff
@@ -20,45 +20,51 @@ <h4 mat-card-title>Note: This is a work in progress for new 'Device Management'
         </div>
         <ix-tree class="disk-tree" [dataSource]="dataSource" [treeControl]="treeControl">
           <ix-tree-node
-            *ixTreeNodeDef="let disk; dataSource: dataSource"
+            *ixTreeNodeDef="let vdev; dataSource: dataSource"
             ixTreeNodeToggle
             [ixTreeNodeToggleRecursive]="false"
             [ixTreeNodeDefDataSource]="dataSource"
-            (click)="onRowSelected(disk, $event)"
-            [class.selected]="disk.guid === selectedItem?.guid"
+            (click)="onRowSelected(vdev, $event)"
+            [class.selected]="vdev.guid === selectedItem?.guid"
           >
             <button mat-icon-button disabled></button>
-            <ix-disk-node [disk]="disk" [size]="diskDictionary[disk.disk]?.size" [type]="diskDictionary[disk.disk]?.type"></ix-disk-node>
+            <ix-disk-node [vdev]="vdev | cast" [disk]="diskDictionary[vdev.disk]"></ix-disk-node>
           </ix-tree-node>
 
           <ix-nested-tree-node
-            *ixTreeNodeDef="let disk; dataSource: dataSource, when: hasNestedChild"
+            *ixTreeNodeDef="let vdev; dataSource: dataSource, when: isVdevGroup"
+            [ixTreeNodeDefDataSource]="dataSource"
+          >
+            <div class="disk-nested-tree-root-node">
+              <ix-vdev-group-node [vdev]="vdev | cast"></ix-vdev-group-node>
```

Comment:
Looks very specific, wouldn't we plan to reuse groups on the dataset tree? I think it may be a part of the ix-tree module. Up to you